modules/dcache/src/main/java/diskCacheV111/poolManager/CostModuleV1.java

```diff
@@ -83,16 +90,61 @@ public ImmutableMap<String, String> getTagMap() {
         public PoolInfo getPoolInfo() {
             return new PoolInfo(_address, _info, _tagMap);
         }
+
+        public long getSerialId() {
+            return _serialId;
+        }
+
+        public int getTrustScore() {
+            return _trustScore;
+        }
+
+        public boolean getEnabledStatus() {
+            return _enabled;
+        }
+    }
+    public boolean getPoolStatus (String poolName) {
+        return _hash.get(poolName).getEnabledStatus();
+
     }
 
     public synchronized void messageArrived(CellMessage envelope, PoolManagerPoolUpMessage msg) {
+        int tsIncrease = 15;
+        int tsDecrease = 2;
+        int tsThreshold = 35;
+        int tsCeiling = 150;
+        long msgSerialId = msg.getSerialId();
+        int nextTrustScore = 0;
+        boolean nextEnabledStatus = true;
+
         CellAddressCore poolAddress = envelope.getSourceAddress();
         String poolName = msg.getPoolName();
         PoolV2Mode poolMode = msg.getPoolMode();
         PoolCostInfo newInfo = msg.getPoolCostInfo();
         Entry poolEntry = _hash.get(poolName);
         boolean isNewPool = poolEntry == null;
 
+        if (!isNewPool) { // Only check for reboots if the pool is not new
+            int lastTrustScore = poolEntry.getTrustScore();
+            long lastSerailId = poolEntry.getSerialId();
+
+            if (msgSerialId == lastSerailId) { // Pool has rebooted
```

Comment:
The comment (or logic) is wrong. If las and new match, then the pool is not rebooted

Reply:
changed in [a51932a](https://github.com/dCache/dcache/pull/7532/commits/a51932a9058dcc27955354b86ef00d2561012e0e)